crates/chain/src/chain_data.rs

```diff
@@ -215,4 +241,64 @@ impl<I: ChainPosition> FullTxOut<I> {
     }
 }
 
+impl<A: Anchor> FullTxOut<ObservedAs<A>> {
+    /// Whether the `txout` is considered mature.
+    ///
+    /// This is the alternative version of [`is_mature`] which depends on `chain_position` being a
+    /// [`ObservedAs<A>`] where `A` implements [`Anchor`].
+    ///
+    /// [`is_mature`]: Self::is_mature
+    pub fn is_mature(&self, tip: u32) -> bool {
+        if !self.is_on_coinbase {
+            return true;
+        }
+
+        let tx_height = match &self.chain_position {
+            ObservedAs::Confirmed(anchor) => anchor.confirmation_height_upper_bound(),
+            ObservedAs::Unconfirmed(_) => {
+                debug_assert!(false, "coinbase tx can never be unconfirmed");
+                return false;
+            }
+        };
+
+        let age = tip.saturating_sub(tx_height);
+        if age + 1 < COINBASE_MATURITY {
+            return false;
+        }
+
+        true
+    }
+
+    /// Whether the utxo is/was/will be spendable with chain `tip`.
+    ///
+    /// This method does not take into account the locktime.
```

Comment:
This seems like a good indicator to put in `locktime` as an optional field in `FullTxOut`? Doesn't have to be this PR, can be added later. Maybe add a TODO here so we don't forget.